### PR TITLE
Fix GitHub release

### PR DESCRIPTION
**Pull request by ludovicsteinbach**
- Fix docs url in pyproject
- Fix condition for dev docs deploy
- Fix gh release step: see https://github.com/pyansys/grantami-bomanalytics/releases/tag/untagged-bb564b6493b96d77faba, generated in CI before putting some conditions back

### CODE REVIEW

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -71,28 +71,43 @@ jobs:
...
-        uses: pyansys/actions/doc-deploy-stable@v4
+      - uses: actions/download-artifact@v3
         with:
-          cname: ${{ env.DOCUMENTATION_CNAME }}
-          token: ${{ secrets.GITHUB_TOKEN }}
-          doc-artifact-name: Documentation-html
+          name: Documentation-html
+          path: ~/dist
+
+      - name: "Compressing HTML documentation"
```
Comment (da1910) [inline .github/workflows/ci_cd.yml:81]: You shouldn't need to explicitely zip artifacts, if you mark a folder for upload then it should be zipped by default

Comment (ludovicsteinbach) [reply .github/workflows/ci_cd.yml:81]: Wouldn't downloading it unzip it then?

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -91,8 +103,11 @@ jobs:
       - name: Release
         uses: softprops/action-gh-release@v1
         with:
+          prerelease: true
```
Comment (Andy-Grigg) [inline .github/workflows/ci_cd.yml]: Presumably these three additions will be removed before merge.

**Review by Andy-Grigg — APPROVED** [review]
All seems good, and the draft release looks right. I don't see anything in the release docs https://github.com/softprops/action-gh-release that explicitly states that it will zip a folder, but it would be nicer if we didn't need to rely on any more actions than absolutely necessary.

File: .github/workflows/ci_cd.yml (hunk 2 above)
Comment (ludovicsteinbach) [reply .github/workflows/ci_cd.yml]: Indeed

**Pull request merged**